### ISSUE TITLE
Stickers from URL

### ISSUE THREAD

**Issue by kesepara** [Jobeso/react-native-whatsapp-stickers#13]
Is it possible to add sticker packs to whatsapp from images which has fetched by url?

If not, is there any solution for this problem ? Maybe saving the image file first etc. ?

**Comment by Jobeso** [comment]
Hey @kesepara,
this is currently not possible and not supported natively by WhatsApp. There are solutions to do that but I am currently not working on it. This is definitely something the library should support in the future. If there is someone willing to support on this, I'm glad for help.

**Comment by kesepara** [comment]
Oh i see. I am working on a project right now that should contains this feature too. So i need to do this on my own. I will let you know when i did that. Do you have any ideas for me to do that quickly? Or any other plugins to work with to upload stickers from URL. For example i think that i need to use react-native-fs first to save the files first. Or do i need to save the files first?

Any suggestions would be really appreciated.

**Comment by Jobeso** [comment]
Yeah I try to give you some hints. First of all, is your project cross platform or only for one os?

**Comment by kesepara** [comment]
> Yeah I try to give you some hints. First of all, is your project cross platform or only for one os?

Cross platform. I've created my project with "react-native init" command.

**Comment by Jobeso** [comment]
So for iOS an approach could be to save them as a local image with rn-fs (like you mentioned) or to base64, convert them to imageData and add support of imageData to this library like discussed in #11. So there is a function `addSticker` in the native code in `StickerPack.swift` with `imageData` type as a parameter. This method is currently not exposed but could be an approach for you. 
For Android it's not that easy and I don't know an approach currently but I can tell you about the problem. On Android the library uses a ContentProvider. This needs to be available even if the app is not in the foreground. A ContentProvider behaves similar to a static method so you can't use the dynamic approach. At least this is what I figured out when developing this library.

**Comment by kesepara** [comment]
> So for iOS an approach could be to save them as a local image with rn-fs (like you mentioned) or to base64, convert them to imageData and add support of imageData to this library like discussed in #11. So there is a function `addSticker` in the native code in `StickerPack.swift` with `imageData` type as a parameter. This method is currently not exposed but could be an approach for you.
> For Android it's not that easy and I don't know an approach currently but I can tell you about the problem. On Android the library uses a ContentProvider. This needs to be available even if the app is not in the foreground. A ContentProvider behaves similar to a static method so you can't use the dynamic approach. At least this is what I figured out when developing this library.

Thank you for hints. I'll let you know if i can make any progress.

**Comment by PasinduDineth** [comment]
waiting for this feature. Thanks for the library. working fine!!

**Comment by srameshr** [comment]
Was this added to the plugin?

**Comment by Jobeso** [comment]
@srameshr No I didn't implement it yet. I am actually thankful for help on that because I don't know a solution how to solve this on Android. 
Do you have plans working on this?

**Comment by srameshr** [comment]
@Jobeso Oh, sorry. I have no idea about it. Does it work on iOS though?

**Comment by Jobeso** [comment]
@srameshr No the functionality is also not supported by default by this library for iOS but I posted a way how to achieve this above. This is only theoretically because the official library from WhatApp doesn't support this behaviour. The above mentioned would be a way I assume it to work on iOS.

**Comment by pureth** [comment]
@Jobeso thanks for taking the time to outline how this could be done on iOS.
The description was useful to me. (as is this entire repo, really awesome!)

I'm just wondering if anything has changed with what you said about Android?

> ...  On Android the library uses a ContentProvider. This needs to be available even if the app is not in the foreground. A ContentProvider behaves similar to a static method so you can't use the dynamic approach. At least this is what I figured out when developing this library.

Any further info here would be greatly appreciated!

Cheers,
Jo

**Comment by Jobeso** [comment]
Hey @pureth,
glad to hear that this library helps you :) 
I browsed the [WhatsApp sticker repo](https://github.com/WhatsApp/stickers) issues lately and found some hints how people implemented this behaviour on the native side but I didn't find the time to dig deeper into it yet.

**Comment by vasconcelosvcd** [comment]
@Jobeso is there any repo that I can see where did you stop so I can try to help you ?

**Comment by Jobeso** [comment]
Hey @vasconcelosvcd,
great to hear that you want to help out, appreciate it. I think the repo you posted is a good lead. They also added some info to their readme about that [here](https://github.com/WhatsApp/stickers/blob/master/Android/README.md#expose-files-that-are-stored-internally-as-stickers-through-contentprovider). 
When I remember correctly my main problem was that the content provider or the content file parser acts statically and I couldn't figure out how to dynamically add stickers through the functions without having them available on app start.

**Comment by ursNj** [comment]
Any update on this ???

**Comment by malithjkmt** [comment]
Has anyone got a PR for this? :)

**Comment by Pcpkomputer** [comment]
up up

**Comment by CrackerKSR** [comment]
I found a guide on medium But I could not use it as I just resume learning android after 3 years
[Part 1](https://medium.com/bobble-engineering/dynamic-whatsapp-stickers-part-1-android-2ac00dc57111)
{Part 2}(https://medium.com/bobble-engineering/how-to-create-dynamic-whatsapp-stickers-part-2-android-b0e3d433984b)

I hope this will help someone who is looking for it
